python-sdk/src/astro/databases/snowflake.py

```diff
@@ -717,11 +708,11 @@ def _build_merge_sql(
         )
 
         merge_target_dict = {
-            f"merge_clause_target_{i}": f"{target_table_name}.{x}"
+            f"merge_clause_target_{i}": f'{target_table_name}."{x}"'
             for i, x in enumerate(target_conflict_columns)
         }
         merge_source_dict = {
-            f"merge_clause_source_{i}": f"{source_table_name}.{x}"
+            f"merge_clause_source_{i}": f'{source_table_name}."{x}"'
```

Comment:
Does quoting fully upper or fully lower values affect those values in snowflake? Like if I go to snowflake do I need to use quotes going forward?

Reply:
@dimberman You can run the query without quotes if we have the cols name in all caps. 
For example - 
If you have below table:
X, Y
1, 2
3,4

case 1:
SELECT x, y From <table_nma> 

case 2:
SELECT X, Y From <table_nma> 

case 3:
SELECT "X", "Y" From <table_nma> 

All of the listed queries will work. 

With the current approach, we are passing every col name in quotes. So that if someone has created cols in upper/lower/mixed cases it will work regardless.

Reply:
but it shouldn't matter, the column names in snowflake are case in-sensitive, no?

Reply:
Nope, doesn't seem like it.
If i have a table with lowercase cols:
x, y
1, 2
3,4

and you execute query 
case 1: 
SELECT x, y From <table_name>
<img width="804" alt="Screenshot 2022-10-14 at 11 16 48 PM" src="https://user-images.githubusercontent.com/13021213/195909279-721840a6-eeed-4c02-b480-f7152b5af073.png">


case 2:
SELECT "x", "y" From <table_name>

<img width="1388" alt="Screenshot 2022-10-14 at 11 17 53 PM" src="https://user-images.githubusercontent.com/13021213/195909426-1dcbd09e-9a90-4501-b247-fae1fad89790.png">

Reply:
The are only case sensitive if you create the table with quotes.

in your screenshot, how are you creating the table? Can you paste the query

Reply:
Exactly, If someone create a table with lower case/Mixed case cols we will be able to handle that.

Reply:
Table creation query - 
CREATE TABLE persons (
    "id" int,
    "name" varchar(255)
);

Reply:
But now we are requiring users to add quotes to all the columns names in their SQL, this is a big con in my opinion

Reply:
So we are keeping them consistent - Whatever user is passing, we are keeping them as it is in the database, and when they are using it in another operator, like data frame they will get the same case for cols.

Reply:
@kaxil They only have to add that if the cols are in lower/mixed case. If they are in uppercase they don't have to add it.

Reply:
Isn't column being in lower case more common?

And I am also talking about the usage in run_raw_sql for example.

as a user now I am forced to add quotes to all my column names, no?

Reply:
yes that's true, it's a tradeoff

## With changes:

### Case 1:
if we insert the below table in snowflake:
X,Y
1, 2,
3, 4

**run_raw_sql():**
we can query it like this:
Select x,y from <table_name>
or 
Select X,Y from <table_name>

**dataframe():**
will return:
X,Y  <-- same as table 
1, 2,
3, 4

### Case 2:
if we insert the below table in snowflake:
x,y
1, 2,
3, 4

**run_raw_sql():**
we can query it like this:
Select "x", "y" from <table_name>

**dataframe():**
will return:
x,y  <-- same as table 
1, 2,
3, 4


## Without Changes

### Case 1:
if we insert the below table in snowflake:
X, Y
1, 2,
3, 4

**run_raw_sql():**
we can query it like this:
Select x,y from <table_name>
or 
Select X,Y from <table_name>

**dataframe():**
will return:
X, Y  <-- same as table 
1, 2,
3, 4

### Case 2:
if we insert the below table in snowflake:
x,y
1, 2,
3, 4

**run_raw_sql():**
we can query it like this:
Select x, y from <table_name>

**dataframe():**
will return:
X,Y  <-- not same as table 
1, 2,
3, 4

Reply:
I did some investigation and I think we should follow `snowflake-sqlalchemy` to resolve this (https://docs.snowflake.com/en/user-guide/sqlalchemy.html#object-name-case-handling) — which is also mentioned in their official docs

My recommendation is we should have the same path as well instead of the current approach  where we use identifiers for everything:

- If data from load_file has “non lower-case” (upper or mixed case), we use “identifiers”
- If data from load_file has “lower_case”, we use it “as-is”

That way we are consistent with snowflake-sqlalchemy  and the most common case of “lower-case” will be catered — i.e. those users remain completely unaffected.

Slack Thread 🧵: https://astronomer.slack.com/archives/C03868KGF2Q/p1666046525734929?thread_ts=1665999175.728989&cid=C03868KGF2Q